code/code/cmd/cmd_whirlwind.cc

```diff
@@ -104,22 +103,25 @@ int TBeing::doWhirlwind()
     damageType = DAMAGE_HACKED;
 
   // Loop for each person in room
-  std::vector<TBeing *> toDelete{};
+  std::vector<TBeing *> validTargets{};
   for (TThing *thing : roomp->stuff) {
     auto *being = dynamic_cast<TBeing *>(thing);
     if (!being || (being == this) || inGroup(*being) ||
         (being->isPc() && IS_SET(desc->autobits, AUTO_NOHARM)) || being->isImmortal() ||
         IS_SET(being->specials.act, ACT_IMMORTAL))
       continue;
 
+    validTargets.push_back(being);
+  }
+
+  for (TBeing *being: validTargets) {
+    if (being->inRoom() != in_room)
```

Comment:
Badly needs a comment.

Reply:
It definitely did, whoops. Added in https://github.com/sneezymud/sneezymud/pull/401/commits/1876906913fa38d5c5c959a560dd97e28a745690.